front/src/applications/stdcm/hooks/useStdcmConsist.ts

```diff
@@ -15,28 +24,53 @@ const useStdcmConsist = () => {
   const totalMass = useSelector(getTotalMass);
   const onTotalMassChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const totalMassValue = Number(e.target.value);
+    setTotalMassChanged(true);
     dispatch(updateTotalMass(totalMassValue === 0 ? undefined : totalMassValue));
   };
 
   const totalLength = useSelector(getTotalLength);
   const onTotalLengthChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const totalLengthValue = Number(e.target.value);
+    setTotalLengthChanged(true);
     dispatch(updateTotalLength(totalLengthValue === 0 ? undefined : totalLengthValue));
   };
 
   const maxSpeed = useSelector(getMaxSpeed);
   const onMaxSpeedChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const totalMaxSpeed = Number(e.target.value);
+    setMaxSpeedChanged(true);
     dispatch(updateMaxSpeed(totalMaxSpeed === 0 ? undefined : totalMaxSpeed));
   };
 
+  const prefillConsist = (
+    rollingStock?: LightRollingStockWithLiveries,
+    towed?: TowedRollingStock
+  ) => {
+    if (!totalMassChanged) {
+      const consistMass = Math.floor(kgToT((rollingStock?.mass ?? 0) + (towed?.mass ?? 0)));
+      dispatch(updateTotalMass(consistMass > 0 ? consistMass : undefined));
+    }
+
+    if (!totalLengthChanged) {
+      const consistLength = Math.floor((rollingStock?.length ?? 0) + (towed?.length ?? 0));
+      dispatch(updateTotalLength(consistLength > 0 ? consistLength : undefined));
+    }
+
+    if (!maxSpeedChanged) {
+      dispatch(
+        updateMaxSpeed(rollingStock?.max_speed ? Math.floor(rollingStock.max_speed) : undefined)
+      );
+    }
```

Comment:
We should leave a comment here to handle this case when the back bug for max speed in towed will be fixed (is that you handling it or another dev ?)

Reply:
Yes, the max speed for towed rolling stock is made in another PR